Library/Homebrew/formula_installer.rb

```diff
@@ -68,6 +68,13 @@ def self.prevent_build_flags
     raise BuildFlagsError.new(build_flags) unless build_flags.empty?
   end
 
+  # Whether a given formula should be built from source during the current
+  # installation run.
+  def self.build_formula_from_source?(f)
+    ARGV.build_all_from_source? ||
+      ((ARGV.build_from_source? || ARGV.build_bottle?) && ARGV.formulae.any? { |argv_f| argv_f.full_name == f.full_name })
```

Comment:
Added an early return to get the line under 80 characters and have single-line expressions. If it's still too long I could add a temporary or something to decompose it further.